api/envoy/config/route/v3/route.proto

```diff
@@ -106,6 +106,19 @@ message RouteConfiguration {
   // option. Users may wish to override the default behavior in certain cases (for example when
   // using CDS with a static route table).
   google.protobuf.BoolValue validate_clusters = 7;
+
+  // The maximum bytes of the response :ref:`direct response body
+  // <envoy_api_field_config.route.v3.DirectResponseAction.body>` size. If not specified the default
+  // is 4096.
+  //
+  // .. warning::
+  //
+  //   Envoy currently holds the content of :ref:`direct response body
+  //   <envoy_api_field_config.route.v3.DirectResponseAction.body>` in memory. Be careful setting
+  //   this to be larger than the default 4KB, since the allocated memory for direct response body
+  //   is not being subject to memory guards.
```

Comment:
I might phrase this as "data plane buffering controls" rather than "memory guards".

Reply:
Updated.